.eslintrc.js

```diff
@@ -835,6 +835,23 @@ module.exports = {
         ],
       },
     },
+    {
+      files: ['**/*.{js,mjs,ts,tsx}'],
+      excludedFiles: ['packages/kbn-i18n/src/react/**/*'],
+      rules: {
+        'no-restricted-imports': [
```

Comment:
This is going to override all the existing instances of `no-restricted-imports`, which isn't good. We should instead add this to module-migration rules in https://github.com/elastic/kibana/blob/master/packages/elastic-eslint-config-kibana/.eslintrc.js#L73-L77 and ignore the error in the specific spot in `kbn-i18n` where we actually need to import `react-intl` with `// eslint-disable-next-line` directives.

Reply:
Oh right, thanks for catching this 🤦‍♂️ 
I wanted to test out if this overrides other rules, but it slipped.

Reply:
@spalger. I noticed that `module-migration` is skipping validating some cases. 
For example, see, how re-export isn't highlighted by this eslint rule: 

https://github.com/Dosant/kibana/blob/15315720dba0446a7fde0405e13413be44759343/packages/kbn-i18n/src/react/inject.tsx#L15

I think `no-restricted-imports` would have caught it. 

I'll file a bug for `module-migration`. But should I continue with it in this pr or should I change to use `no-restricted-imports`?

Reply:
filed a bug: https://github.com/elastic/kibana/issues/102603